test/FeeCollector.t.sol

```diff
@@ -0,0 +1,539 @@
+        uint256 postOwnerBalance = CONTRACT_DEPLOYER.balance;
+
+        // Assertions
+        assertEq(postContractBalance, 0);
+        assertEq(postOwnerBalance, preOwnerBalance + _amount);
+    }
+
+    /// @dev
+    // - It should revert with Unauthorized() error when called by an unauthorized sender.
+    function testFuzz_CannotExtractNative(uint256 _amount, address _sender) public {
+        // Setup: fund contract with _amount of native token
+        _amount = bound(_amount, 1, 1e22);
+        vm.assume(_sender != CONTRACT_DEPLOYER);
+        vm.deal(feeCollectorAddr, _amount);
+
+        // Act: attempt to extract native
+        vm.prank(_sender);
+        vm.expectRevert(abi.encodeWithSelector(OwnableUnauthorizedAccount.selector, _sender));
+        feeCollector.extractNative();
+    }
+
+    /// @dev
+    // - The FeeCollector's token balance should decrease by its total balance - totalClientBalances
+    // - The owner's token balance should increase by the amount transferred.
+    // - The token totalClientBalances should not change
+    function testFuzz_ExtractERC20(uint256 _amount) public {
+        for (uint256 i; i < positions.length; i++) {
+            // Test Variables
+            address token = positions[i].cToken;
+
+            // Bound fuzzed variables
+            _amount = bound(_amount, assets.minCAmts(token), assets.maxCAmts(token));
+
+            // Fund positionOwner with _amount of token
+            _fund(positionOwner, token, _amount);
+
+            // Approve FeeCollector contract to spend token
+            IERC20(token).approve(feeCollectorAddr, _amount);
+
+            // Collect fees
+            feeCollector.collectFees(TEST_CLIENT, token, _amount);
+
+            // Pre-act balances
+            uint256 preContractTokenBalance = IERC20(token).balanceOf(feeCollectorAddr);
+            uint256 preOwnerTokenBalance = IERC20(token).balanceOf(CONTRACT_DEPLOYER);
+            uint256 preTotalClientBalances = feeCollector.totalClientBalances(token);
+
+            // Act: owner withraws fees
+            vm.prank(CONTRACT_DEPLOYER);
+            feeCollector.extractERC20(token);
+
+            // Post-act balances
+            uint256 postContractTokenBalance = IERC20(token).balanceOf(feeCollectorAddr);
+            uint256 postOwnerTokenBalance = IERC20(token).balanceOf(CONTRACT_DEPLOYER);
+            uint256 postTotalClientBalances = feeCollector.totalClientBalances(token);
+
+            // Assertions
+            assertEq(postContractTokenBalance, preTotalClientBalances);
+            assertEq(postOwnerTokenBalance, preOwnerTokenBalance + (preContractTokenBalance - preTotalClientBalances));
+            assertEq(postTotalClientBalances, preTotalClientBalances);
+        }
+    }
+
+    /// @dev
+    // - It should revert with Unauthorized() error when called by an unauthorized sender.
+    function testFuzz_CannotExtractERC20(uint256 _amount, address _sender) public {
```

Comment:
`_amount` -> `_protocolFee`.

Reply:
Keeping as amount, since this function is not limited to extracting protocol fees.